juniper_official/Solutions/Icmp-outlier/icmp.rule

```diff
@@ -0,0 +1,252 @@
+                 */
+                term is-rtt-medium {
+                    when {
+                        equal-to "$dt-response-time" 1;
+                    }
+                    then {
+                        status {
+                            color yellow;
+                            message "Round trip time($rtt-average μs) to $host is above dynamic threshold";
+                        }
+                    }
+                }
+                /*
+                 * Defaults color to green.
+                 */
+                term rtt-normal {
+                    then {
+                        status {
+                            color green;
+                            message "Round trip time($rtt-average μs) to $host is normal";
+                        }
+                    }
+                }
+            }
+            variable count_var {
+                value 1;
+                type sensor-argument;
+                description "Input ICMP probe count";
+            }
+            variable host_var {
+                value "[null]";
+                type sensor-argument;
+                description "Input Destination host";
+            }
+            variable rtt-threshold-var {
+                value 1000;
+                description "Input RTT response static threshold in μs";
+                type int;
+            }
+        }
+        rule check-outlier {
+            description "Detects ICMP outlier response time and notify anomalies";
+            synopsis "ICMP outlier analyzer";
+            rule-frequency 60s;
+            network-rule;
+            field rtt-ol {
+                formula {
+                    outlier-detection {
+                        algorithm {
+                            dbscan {
+                                sensitivity {
+                                    level low;
+                                }
+                                learning-period 120s;
+                            }
+                        }
+                        dataset "{{rtt_xpath}}";
+                    }
+                }
+                type string;
+                description "Outlier detection of devices in a group using xpath";
+            }
+            trigger icmp-outlier-detection {
+                synopsis "ICMP outlier KPI";
+                description "Sets health based on the RTT average response time across devices";
+                frequency 60s;
+                /*
+                 * Sets color to red and sends out an anomaly notification
+                 * when outlier detected by algorithm across devices.
+                 */
+                term is-outlier-detected {
+                    when {
+                        equal-to "$rtt-ol-is-outlier" 1;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Outlier detected on $rtt-ol-device";
```

Comment:
Can we change "$rtt-ol-device" to something more descriptive? It took me a while to figure out that ol referred to outlier.

Reply:
It is not allowing more characters hence used this way